sdk/core/azure-core/inc/azure/core/internal/system_clock.hpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <functional>
+
+#include "azure/core/dll_import_export.hpp"
+#include "azure/core/http/transport.hpp"
+
+namespace Azure { namespace Core { namespace _internal {
+  class SystemClock {
```

Comment:
This is a lot of complexity to mock the system clock.  Can you describe the scenario where we want to replace the system clock on the fly.

Reply:
How else would I have an expectation for token expiration value?

Reply:
Going forward, I'll use this to test retry policy, and token refresh (BearerTokenAuthenticationPolicy)

Reply:
It would be simpler to back date the expiration time on the token rather than artificially moving the system clock forward/backward.

Reply:
Token does not have the expiration date, it has "expires_in", which is in seconds from now. And we can't be precise about "now". Even if I get timestamp for before and after, and verify that the expiration falls between them, very soon I'll need to write a test for BererTokenAuthenticationPolicy and RetryPolicy, where we'll need to inject the values for DateTime.Now + Thread.Sleep.